index.js

```diff
@@ -138,6 +138,9 @@ AFRAME.registerComponent('teleport-controls', {
     this.hitEntity.setAttribute('visible', false);
 
     if ('collisionEntities' in diff) { this.queryCollisionEntities(); }
+    if (!this.raycastPoints) {
```

Comment:
I'd create `this.raycastPoints` in `init` (where you don't have to check if `this.raycastPoints` exists), and recreate it whenever `'curveNumberPoints' in diff`, since when that number changes you want to have a different number of raycastPoints.